ibm/resource_ibm_private_dns_glb.go

```diff
@@ -327,3 +334,34 @@ func suppressPDNSGlbNameDiff(k, old, new string, d *schema.ResourceData) bool {
 	}
 	return false
 }
+
+func isWaitForLoadBalancerDeleted(LoadBalancer *dnssvcsv1.DnsSvcsV1, d *schema.ResourceData, timeout time.Duration) (interface{}, error) {
+	idset := strings.Split(d.Id(), "/")
+	log.Printf("Waiting for PDNS GLB (%s) to be deleted.", idset[2])
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"retry", pdnsGLBDeleting},
+		Target:     []string{pdnsGLBDeleted, ""},
+		Refresh:    isVLoadBalancerDeleteRefreshFunc(LoadBalancer, d),
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 10 * time.Second,
+	}
+
+	return stateConf.WaitForState()
+}
+
+func isVLoadBalancerDeleteRefreshFunc(LoadBalancer *dnssvcsv1.DnsSvcsV1, d *schema.ResourceData) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		idset := strings.Split(d.Id(), "/")
+		deletelbOptions := LoadBalancer.NewDeleteLoadBalancerOptions(idset[0], idset[1], idset[2])
+		response, err := LoadBalancer.DeleteLoadBalancer(deletelbOptions)
```

Comment:
you should call `GetLoadBalancer` method here.

Reply:
Handled